inline.h

```diff
@@ -455,6 +455,414 @@ S_sv_or_pv_pos_u2b(pTHX_ SV *sv, const char *pv, STRLEN pos, STRLEN *lenp)
+      copy_length(XPVAV, xav_alloc),
+      0,
+      SVt_PVAV, TRUE, NONV, HASARENA,
+      FIT_ARENA(0, sizeof(ALIGNED_TYPE_NAME(XPVAV))) },
+
+    { sizeof(ALIGNED_TYPE_NAME(XPVHV)),
+      copy_length(XPVHV, xhv_max),
+      0,
+      SVt_PVHV, TRUE, NONV, HASARENA,
+      FIT_ARENA(0, sizeof(ALIGNED_TYPE_NAME(XPVHV))) },
+
+    { sizeof(ALIGNED_TYPE_NAME(XPVCV)),
+      sizeof(XPVCV),
+      0,
+      SVt_PVCV, TRUE, NONV, HASARENA,
+      FIT_ARENA(0, sizeof(ALIGNED_TYPE_NAME(XPVCV))) },
+
+    { sizeof(ALIGNED_TYPE_NAME(XPVFM)),
+      sizeof(XPVFM),
+      0,
+      SVt_PVFM, TRUE, NONV, NOARENA,
+      FIT_ARENA(20, sizeof(ALIGNED_TYPE_NAME(XPVFM))) },
+
+    { sizeof(ALIGNED_TYPE_NAME(XPVIO)),
+      sizeof(XPVIO),
+      0,
+      SVt_PVIO, TRUE, NONV, HASARENA,
+      FIT_ARENA(24, sizeof(ALIGNED_TYPE_NAME(XPVIO))) },
+};
+
+
+#ifndef PURIFY
+
+/* grab a new thing from the arena's free list, allocating more if necessary. */
+#define new_body_from_arena(xpv, root_index, type_meta) \
+    STMT_START { \
+        void ** const r3wt = &PL_body_roots[root_index]; \
+        xpv = (PTR_TBL_ENT_t*) (*((void **)(r3wt))      \
+          ? *((void **)(r3wt)) : Perl_more_bodies(aTHX_ root_index, \
+                                             type_meta.body_size,\
+                                             type_meta.arena_size)); \
+        *(r3wt) = *(void**)(xpv); \
+    } STMT_END
+
+PERL_STATIC_FORCE_INLINE void *
+S_new_body(pTHX_ const svtype sv_type)
+{
+    void *xpv;
+    new_body_from_arena(xpv, sv_type, bodies_by_type[sv_type]);
+    return xpv;
+}
+
+#endif
+
+#define new_body_allocated(sv_type)             \
+    (void *)((char *)S_new_body(aTHX_ sv_type)  \
+             - bodies_by_type[sv_type].offset)
+
+#ifdef PURIFY
+#if !(NVSIZE <= IVSIZE)
+#  define new_XNV()     safemalloc(sizeof(XPVNV))
+#endif
+#define new_XPVNV()     safemalloc(sizeof(XPVNV))
+#define new_XPVMG()     safemalloc(sizeof(XPVMG))
+
+#define del_body_by_type(p, type)       safefree(p)
+
+#else /* !PURIFY */
+
+#if !(NVSIZE <= IVSIZE)
+#  define new_XNV()     new_body_allocated(SVt_NV)
+#endif
+#define new_XPVNV()     new_body_allocated(SVt_PVNV)
+#define new_XPVMG()     new_body_allocated(SVt_PVMG)
+
+#define del_body_by_type(p, type)                               \
+    del_body(p + bodies_by_type[(type)].offset,                 \
+             &PL_body_roots[(type)])
+
+#endif /* PURIFY */
+
+
+/* no arena for you! */
+
+#define new_NOARENA(details) \
+        safemalloc((details)->body_size + (details)->offset)
+#define new_NOARENAZ(details) \
+        safecalloc((details)->body_size + (details)->offset, 1)
+
+
+/*
+=for apidoc sv_upgrade_fresh
+
+A cut-down version of sv_upgrade intended only for when sv is a
+freshly-minted sv that hasn't been upgraded or otherwised used.
+
+Changes should only be made to this function to reflect changes
+to the main Perl_sv_upgrade function (in sv.c).
+
+=cut
+*/
+
+#define PERL_IN_SV_C
```

Comment:
This effectively defines `PERL_IN_SV_C` **everywhere**, which defeats its purpose. If you need some symbol that isn't available otherwise, you need to make it public.